### ISSUE TITLE
Code is incorrectly marked as unreachable when using nullish coalescing

### ISSUE THREAD

**Issue by tomaskikutis** [microsoft/TypeScript#34635]
**TypeScript Version:**  3.7.0-beta

**Search Terms:**

Unreachable code detected

**Code**

```ts
const maybeBool = false;
if (!(maybeBool ?? true)) {
    console.log('...'); // <-- marked as unreachable
}
```

**Playground Link:** 

https://www.typescriptlang.org/play/index.html?ts=3.7-Beta&ssl=5&ssc=1&pln=1&pc=1#code/MYewdgzgLgBAtgQwJ4CMCmAhEIA2MC8MAZgjhGgNwBQAlkTABQCEDiqm2eA-FzFAE4BXNAEoRMAN5UYMmKEi40AOhwgA5gwDkSnZpHUAvlSA

**Comment by ahejlsberg** [comment]
Additional repros:

```ts
declare function foo(): void;

const maybeBool = false;

if (!(maybeBool ?? true)) {
    foo();  // Unreachable
}

if (maybeBool ?? true) {
    foo();
}
else {
    foo();  // Unreachable
}

if (false ?? true) {
    foo();
}
else {
    foo();  // Unreachable
}
```

The unreachability errors above are wrong.
